### PR TITLE
[Camel] Cannot execute TypeScript files

### PR DESCRIPTION
**Pull request by ThuF**
Contribution PR:
- https://github.com/eclipse/dirigible/pull/3395

### CODE REVIEW

File: modules/engines/engine-graalium/execution-core/src/main/java/org/eclipse/dirigible/graalium/core/DirigibleJavascriptCodeRunner.java [hunk 1]
```diff
@@ -178,8 +182,7 @@ public Module run(Path codeFilePath) {
 
     public Value runMethod(Module codeModule, String methodName, Object... args) {
         return switch (codeModule.moduleType()) {
-            case CJS -> runCjsMethod(codeModule.module(), methodName, args);
-            case ESM -> runEsmMethod(codeModule.module(), methodName, args);
+            case CJS, ESM -> runEsmMethod(codeModule.module(), methodName, args);
```
Comment (vmutafov) [inline modules/engines/engine-graalium/execution-core/src/main/java/org/eclipse/dirigible/graalium/core/DirigibleJavascriptCodeRunner.java:185]: Why is this necessary?

File: modules/engines/engine-graalium/execution-core/src/main/java/org/eclipse/dirigible/graalium/core/DirigibleJavascriptCodeRunner.java [hunk 2]
```diff
@@ -241,10 +236,7 @@ public GraalJSInterceptor getGraalJSInterceptor() {
     }
 
     private static String transformTypeScriptHandlerPathIfNecessary(String handlerPath) {
-        if (handlerPath.endsWith(TS_EXT)) {
-            return handlerPath.substring(0, handlerPath.length() - TS_EXT.length()) + MJS_EXT;
-        }
-        return handlerPath;
+        return handlerPath.endsWith(TS_EXT) ? handlerPath.replaceAll(Pattern.quote(TS_EXT), JS_EXT) : handlerPath;
```
Comment (vmutafov) [inline modules/engines/engine-graalium/execution-core/src/main/java/org/eclipse/dirigible/graalium/core/DirigibleJavascriptCodeRunner.java:239]: This is a breaking change in some cases. Why are we changing the extension to “.js”?

File: modules/engines/engine-graalium/execution-core/src/main/java/org/eclipse/dirigible/graalium/core/DirigibleJavascriptCodeRunner.java (hunk 1 above)
Comment (vmutafov) [reply modules/engines/engine-graalium/execution-core/src/main/java/org/eclipse/dirigible/graalium/core/DirigibleJavascriptCodeRunner.java]: This is incorrect. If there is a problem somewhere, it’s definitely not here. CommonJS and ESM are two completely different module systems and there is no standard way for interoperability between them. We shouldn’t treat CommonJS modules as ESMs.

**Review by vmutafov — CHANGES_REQUESTED** [review]

File: modules/engines/engine-graalium/execution-core/src/main/java/org/eclipse/dirigible/graalium/core/DirigibleJavascriptCodeRunner.java (hunk 1 above)
Comment (ThuF) [reply modules/engines/engine-graalium/execution-core/src/main/java/org/eclipse/dirigible/graalium/core/DirigibleJavascriptCodeRunner.java]: Reverted to the original content and execution flow

File: modules/engines/engine-graalium/execution-core/src/main/java/org/eclipse/dirigible/graalium/core/DirigibleJavascriptCodeRunner.java (hunk 2 above)
Comment (ThuF) [reply modules/engines/engine-graalium/execution-core/src/main/java/org/eclipse/dirigible/graalium/core/DirigibleJavascriptCodeRunner.java:248]: With the latest commit the file extension is still changed from `*.ts` to `*.js`, however in the case of `*.ts` the **ModuleType** is assigned to `ModuleType.ESM`

**Review by vmutafov — APPROVED** [review]
